resolwe_bio/processes/import_data/bwamem2_index_files.py

```diff
@@ -0,0 +1,75 @@
+"""Upload index files for BWA-MEM2 aligner"""
+import gzip
+import shutil
+from pathlib import Path
+
+from resolwe.process import (
+    Cmd,
+    DataField,
+    DirField,
+    FileField,
+    Process,
+    SchedulingClass,
+    StringField,
+)
+
+
+class ImportBWA2Index(Process):
+    """Import BWA-MEM2 index files."""
+
+    slug = "upload-bwamem2-index"
+    name = "BWA-MEM2 index files"
+    process_type = "data:index:bwamem2"
+    version = "1.0.0"
+    category = "Import"
+    scheduling_class = SchedulingClass.BATCH
+    requirements = {
+        "expression-engine": "jinja",
+        "executor": {
+            "docker": {"image": "public.ecr.aws/genialis/resolwebio/dnaseq:6.1.0"},
+        },
+    }
+    data_name = '{{ ref_seq.fasta.file|basename|default("?") }}'
+    version = "1.0.0"
+
+    class Input:
+        """Input fields to process Import BWA-MEM2 index."""
+
+        ref_seq = DataField(
+            "seq:nucleotide", label="Reference sequence (nucleotide FASTA)"
+        )
+        index_name = FileField(label="BWA-MEM2 index files")
+
+    class Output:
+        """Output fields to process BWAMEM2Index."""
+
+        index = DirField(label="BWA-MEM2 index")
+        fastagz = FileField(label="FASTA file (compressed)")
+        fasta = FileField(label="FASTA file")
+        fai = FileField(label="FASTA file index")
+        species = StringField(label="Species")
+        build = StringField(label="Build")
+
+    def run(self, inputs, outputs):
+        """Run the analysis"""
+        basename = Path(inputs.ref_seq.output.fasta.path).name
+        assert basename.endswith(".fasta")
+        name = basename[:-6]
+
+        index_dir = Path("BWAMEM2_index")
+        if not index_dir.exists():
+            index_dir.mkdir()
+
+        index_file = inputs.index_name.import_file(imported_format="compressed")
+        shutil.unpack_archive(index_file, index_dir)
+
+        shutil.copy(Path(inputs.ref_seq.output.fasta.path), Path.cwd())
```

Comment:
When this is not the existing Data object, you need to handle the import differently. For FASTA import procedure, you can look at the existing code in the process `upload-fasta-nucl`